articles/debug-node-devtools/index.md

```diff
@@ -0,0 +1,113 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/debug-node-devtools/
+title: Debugging a Node.Js app using Chrome Dev Tools
+description: Learn how to set up a local and remote app for debugging. For remote apps, we set up an SSH tunnel for debugging. The article addresses how opening DevTools, setting breakpoints, and restarting app instances for debugging. 
+author: geoffrey-mungai
+date: 2020-08-24T00:00:00-10:00
+topics: [Node.js]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/debug-node-devtools/hero.jpg
+    alt: 
+---
+Node.js is a robust technology stack with a rich ecosystem. However, debugging Node.js applications remains a difficult task, since they can fail without any error message. Using Chrome Dev Tools is a simple solution to finding errors in your code.
+<!--more-->
+### Introduction
+
+Node.js is a fast, asynchronous and event driven server-side JavaScript technology. You can read [why Node.js is popular among developers](https://www.section.io/engineering-education/why-node-js-is-popular/). 
+
+A debugger is a program that is used to inspect a program as it runs. It enables running the code line-by-line and following the execution process. In the process, you can see the code as a flow chart and logically interpret it. This enables you to find bugs more easily.
+
+The most popular way JavaScript and Node developers track bugs in their programs is printing data to the console. They use methods like [console.log()](https://developer.mozilla.org/en-US/docs/Web/API/Console/log) and [console.debug()](https://developer.mozilla.org/en-US/docs/Web/API/Console/debug). Although these methods can be helpful, they require a lot of manual effort and are not foolproof. Suppose you have a large app, how many `console.log()` statements would you need? Hundreds, perhaps. 
+
+Logging sensitive information can occur, exposing your app to malicious exploitations and data breaches. Debuggers come handy in a situation like this. They help track bugs without exposing your app to security threats.
+
+Sometimes Node.js apps can fail without error messages. This occurs mainly in cases where [error handling](https://nodejs.dev/learn/error-handling-in-nodejs) is not employed. Debuggers can watch behaviors of specific objects throughout program execution by adding breakpoints in a program. Breakpoints are points in your code where you pause execution and examine various factors. By watching objects, you can evaluate their values and type as the execution goes on. 
```

Comment:
Even so, Node.js apps can fail without error messages.

Reply:
Breakpoints are locations in your code where you pause execution and examine various factors.